include/power.h

```diff
@@ -27,8 +27,8 @@ enum power_states {
 	SYS_POWER_STATE_AUTO	= (-2),
 	SYS_POWER_STATE_ACTIVE	= (-1),
 #ifdef CONFIG_SYS_POWER_LOW_POWER_STATES
-# ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_SUPPORTED
-	SYS_POWER_STATE_CPU_LPS,
+# ifdef CONFIG_SYS_POWER_STATE_CPU_LPS_0_SUPPORTED
+	SYS_POWER_STATE_CPU_LPS_0,
```

Comment:
There is no problem to change that. We just have to decide what would be best.

Reply:
I disagree there. I think that power states starting with 0 make sense in this context, because they are mappable to indices in an array. In the Bluetooth world for example we always have packet types starting from 0, or connection types starting from 0.

Reply:
In this case the numerical representation of the state will have no connection to its name. It's best to look at the definition of `enum power_states` in power.h. Having a name suggests that such connection exists is potentially dangerous.

While vendors typically provide multiple low power / deep sleep levels they never start naming them with index 0, if the indexes are used at all. That's not natural in this use case. 0 is in fact often reserved to mean an active state.

Reply:
> is potentially dangerous.

I apologize for my poor English. I was writing in a hurry. There is nothing dangerous about having names starting with index 0 but it is misleading. Here are a few examples of state mapping for low power states:
- quark_se_c1000_ss: LPS_0 -> SS1, LPS_1 -> SS2
- intel_quark: LPS_0 -> C1, LPS_1 -> C2, LPS_2 -> C2LP 
- SiLabs: LPS_0 -> EM1 Sleep, LPS_1 -> EM2 DeepSleep (active state is called EM0 Active)

assuming that we proceed with the current naming convention.

Also we need to take into account that the Kernel itself puts CPU in the sleep mode by calling k_cpu_idle(). This is also a low power mode typically documented in the datasheet in the section on power management. On ARM architecture this is a mode entered by calling WFI instruction with no need for any interaction with SoC power management module. It turns off CPU clock. This is in addition to LPS_0 which is the first 'real' low power state. So we end up with one sub 0 low power state.

Reply:
I have renamed the states. Now they are starting from 1.
This better match the well known C0, C1, Cn naming where C0 is the active state.